tests/conftest.py

```diff
@@ -764,8 +765,13 @@ def creds_on_dut(duthost):
 
     creds["ansible_altpasswords"] = []
 
+    if is_ip_reachable(duthost, duthost.mgmt_ip):
+        dutip = duthost.mgmt_ip
+    else:
+        dutip = duthost.mgmt_ipv6
+
     passwords = creds["ansible_altpasswords"] + [creds["sonicadmin_password"]]
-    creds['sonicadmin_password'] = get_dut_current_passwd(duthost.mgmt_ip, creds['sonicadmin_user'], passwords)
+    creds['sonicadmin_password'] = get_dut_current_passwd(dutip, creds['sonicadmin_user'], passwords)
```

Comment:
I would suggest enhancing get_dut_current_passwd, to receive both IPv4 ip and IPv6 ip and internally retry.
An not that similar example: https://github.com/sonic-net/sonic-mgmt/blob/3be50896eeff945f32d6895fcbbaaf935017e713/tests/common/fixtures/duthost_utils.py#L677

Reply:
I updated get_dut_current_passwd, please help re-review